shell/platform/darwin/macos/framework/Headers/FlutterPluginMacOS.h

```diff
@@ -22,7 +23,7 @@ NS_ASSUME_NONNULL_BEGIN
  * expand over time to more closely match the functionality of the iOS FlutterPlugin.
  */
 FLUTTER_DARWIN_EXPORT
-@protocol FlutterPlugin <NSObject>
+@protocol FlutterPlugin <NSObject, FlutterAppLifecycleDelegate>
```

Comment:
This is to match iOS. Personally I would have not done this, and had plugins that also wanted to be `FlutterAppLifecycleDelegate`s implement this themselves, but the ship has sailed for iOS and there's not really much downside, so I'm opting for consistency with iOS (as I've tried to do in plugin APIs in general, to minimize pain in code sharing in plugins).

Reply:
Agreed - getting the headers as similar as possible both for consistency and with an eye to a future world in which we can merge the non-UIKit/AppKit specific parts implies that we've got to live with the union of past decisions across both that we might prefer to not have made in retrospect.